src/Mod/Sketcher/Gui/EditModeInformationOverlayCoinConverter.cpp

```diff
@@ -82,167 +86,191 @@ void EditModeInformationOverlayCoinConverter::addToInfoGroup(SoSwitch * sw) {
 
 template < EditModeInformationOverlayCoinConverter::CalculationType calculation >
 void EditModeInformationOverlayCoinConverter::calculate(const Part::Geometry * geometry, [[maybe_unused]] int geoid) {
-    const Part::GeomBSplineCurve *spline = static_cast<const Part::GeomBSplineCurve *>(geometry);
-
-    if constexpr (calculation == CalculationType::BSplineDegree ) {
-        clearCalculation(degree);
 
-        std::vector<Base::Vector3d> poles = spline->getPoles();
+    if (geometry->getTypeId() == Part::GeomBSplineCurve::getClassTypeId()){

```

Comment:
Would suggest turning this into a guard clause (flipping the condition and moving the else lines in here.)

Reply:
The ArcOfConic case could be treated next (to show an elliptic helper) and the `else` statement will become `else if (geometry->getTypeId() == Part::GeomArcOfCircle::getClassTypeId())`

It could be better to add a default failing case with `assert(false /*not implemented*/)`.

Reply:
Ok, got it !